packages/sdk-effects/storage/src/lib/storage.effects.ts

```diff
@@ -31,109 +27,103 @@ export interface CustomStorageConfig {
   custom: CustomStorageObject;
 }
 
-export function createStorage<Value>(config: StorageConfig) {
+function createStorageError(
+  storeType: 'localStorage' | 'sessionStorage' | 'custom',
+  action: 'Storing' | 'Retrieving' | 'Removing' | 'Parsing',
+): GenericError {
+  let storageName;
+  switch (storeType) {
+    case 'localStorage':
+      storageName = 'local';
+      break;
+    case 'sessionStorage':
+      storageName = 'session';
+      break;
+    case 'custom':
+      storageName = 'custom';
+      break;
+    default:
+      break;
+  }
+
+  return {
+    error: `${action}_error`,
+    message: `Error ${action.toLowerCase()} value from ${storageName} storage`,
+    type: action === 'Parsing' ? 'parse_error' : 'unknown_error',
+  };
+}
+
+export function createStorage<Value>(config: StorageConfig): StorageClient<Value> {
   const { type: storeType, prefix = 'pic', name } = config;
+  const key = `${prefix}-${name}`;
+  const storageTypes = {
+    sessionStorage,
+    localStorage,
+  };
 
   if (storeType === 'custom' && !('custom' in config)) {
     throw new Error('Custom storage configuration must include a custom storage object');
   }
 
-  const key = `${prefix}-${name}`;
   return {
     get: async function storageGet(): Promise<Value | GenericError | null> {
-      if ('custom' in config) {
+      if (storeType === 'custom') {
         const value = await config.custom.get(key);
-        if (value === null) {
+        if (value === null || (typeof value === 'object' && 'error' in value)) {
```

Comment:
I'm a bit confused by this. If `custom.get` can only return a `string` or `null`, why are we checking for objects and error properties?